uint8[] data        # Serialized encrypted data as byte array
uint32 data_type   # 0: x coordinate, 1: y coordinate, 2: result 